service/storage_service.cc

```diff
@@ -5408,11 +5410,12 @@ void storage_service::init_messaging_service(sharded<service::storage_proxy>& pr
                 curr_cdc_gen_id = ss._topology_state_machine._topology.current_cdc_generation_id;
             }
 
-            std::optional<canonical_mutation> cdc_generation_mutation;
+            std::vector<canonical_mutation> cdc_generation_mutations;
             if (curr_cdc_gen_id) {
                 // We don't need to fetch this data under group0 apply mutex, it's immutable.
```

Comment:
"immutable" was true when referring to the current CDC generation which we would only transfer once it was fully applied.

Now we transfer the entire contents of CDC_GENERATIONS_V3, which may contain a partially applied generation data which we are concurrently applying more of. So it is no longer immutable.

However, it may be that it is still safe to pull such generations that are being concurrently applied -- but now the reason is different. If you found such a reason and convinced yourself that it is safe, then you should also convince us (by updating the comment appropriately).

Reply:
I'm not 100% sure it is safe, so I will add this mutex here.

Reply:
Fixed.